src/infrastructure/events/OnCommentAdded.ts

```diff
@@ -0,0 +1,47 @@
+import { ApolloClient } from 'apollo-boost'
+
+import { AppEventListener } from '../../domain/events/Dispatcher'
+
+import { queries } from '../repository/SpotRepositoryGql'
+import {
+  FetchSpot,
+  FetchSpotVariables,
+} from '../repository/queries/__generated__/FetchSpot'
+
+export interface Dependencies {
+  apollo: ApolloClient<any>
+}
+
+export const OnCommentAdded = ({
+  apollo,
+}: Dependencies): AppEventListener => event => {
+  if (event.type === 'comment_added') {
```

Comment:
Will we have to filter the event type in all the listeners?

On a side note, an early return would ease the comprehension.

Reply:
Yep we'll have to, our we can improve the API to only listen to given types.

Side note - an early return will work in this PR (as `AppEvent` is only an alias to 1 event type); but won't as soon as we have multiple types if we do not exclude every other type in the if.